### PR TITLE
Make dataset methods optional

### PR DESCRIPTION
**Pull request by andrewgryan**
# Optional map_view and navigator

Optional methods

## Check list

Handy reminders of things to do ahead of merge

- [ ] Bump version number to reflect change, edit `forest/__init__.py`
  <sub>(Use semantic version x.y.z where x is major, y is feature, z is patch)</sub>

### CODE REVIEW

File: forest/main.py [hunk 1]
```diff
@@ -183,12 +187,16 @@ def is_image(renderer):
...
+    sub_navigators = {
+        key: dataset.navigator() for key, dataset in datasets.items()
+        if hasattr(dataset, "navigator")
+    }
+    navigator = navigate.Navigator(sub_navigators)
 
     # Pre-select menu choices (if any)
     initial_state = {}
-    for _, pattern in config.patterns:
+    for pattern, _ in sub_navigators.items():
```
Comment (twsearle) [inline forest/main.py]: I am a bit confused here. The keys in `sub_navigators` are now the patterns for db.initial_date, but it looks like they are in fact Navigators returned by `dataset.navigator()`. Can't quite tell why this all works!

File: forest/main.py [hunk 2]
```diff
@@ -92,15 +92,18 @@ def main(argv=None):
...
-        renderers[group.label] = [
-                viewer.add_figure(f)
-                for f in figures]
+
+        # Add optional map view
+        if hasattr(dataset, "map_view"):
+            map_view = dataset.map_view()
+            map_views[group.label] = map_view
+            renderers[group.label] = [
+                    map_view.add_figure(f)
```
Comment (twsearle) [inline forest/main.py]: Maybe not for this one, but perhaps `map_view.add_figure` should be called `map_view.add_renderer`?

File: forest/main.py (hunk 1 above)
Comment (andrewgryan) [reply forest/main.py]: @twsearle yes this definitely needs to change in a future PR, the keys for the Navigator should be a unique ID to reference sub-navigators

Comment (andrewgryan) [reply forest/main.py]: @twsearle also, perhaps it's time for `db.initial_state()` to move somewhere else?

**Pull request merged**